services/placement/service/placementservice.go

```diff
@@ -91,6 +84,33 @@ func (ps placementService) BuildInitialPlacement(
 	return p, ps.ss.SetIfNotExist(ps.service, p)
 }
 
+func (ps placementService) initPlacement(
+	instances []services.PlacementInstance,
+	ids []uint32,
+	rf int,
+) (services.ServicePlacement, error) {
+	if (rf == 0) == ps.opts.Sharded() {
+		return nil, fmt.Errorf("could not build initial placement with rf = %v and sharded = %v", rf, ps.opts.Sharded())
+	}
+
+	p, err := ps.algo.InitialPlacement(instances, ids)
```

Comment:
the algo will be the non-sharded one when rf = 0, but I'm going to remove the relation between rf and IsSharded() as we discussed